mlir/lib/Analysis/DataFlow/DenseAnalysis.cpp

```diff
@@ -139,15 +139,37 @@ void AbstractDenseDataFlowAnalysis::visitRegionBranchOperation(
   assert(predecessors->allPredecessorsKnown() &&
          "unexpected unresolved region successors");
 
+  auto getBeforeBranchState = [&]() -> const AbstractDenseLattice * {
+    const AbstractDenseLattice *before;
+    if (Operation *prev = branch->getPrevNode())
+      before = getLatticeFor(point, prev);
+    else
+      before = getLatticeFor(point, branch->getBlock());
+    return before;
+  };
+
+  if (auto *op = point.dyn_cast<Operation *>())
+    if (op == branch)
+      // If the number of known predecessors is one less than the number of
```

Comment:
"predecessors" is a well know term in compiler theory. What are the predecessors here ? the predecessors blocks of the if branch ? The predecessors of the instructions after the if branch. This comment needs a lot more explanation.

Reply:
changed the comment.